soccer/LogViewer.cpp

```diff
@@ -1,37 +1,82 @@
 #include <LogViewer.hpp>
+#include <NewRefereeModule.hpp>
 
 #include <google/protobuf/io/zero_copy_stream_impl.h>
+#include <protobuf/LogFrame.pb.h>
+#include <protobuf/referee.pb.h>
 
 #include <QApplication>
 #include <QFile>
+#include <QTextStream>
 
 #include <algorithm>
 #include <fcntl.h>
-#include <stdio.h>
+#include <time.h>
+
+#include <iostream>
 
 using namespace std;
 using namespace boost;
 using namespace Packet;
 using namespace google::protobuf::io;
+using namespace google::protobuf;
 
+/**
+ * Defines usage information for launching the Log-Viewer application
+ * @param prog The name of the program
+ */
 void usage(const char* prog) {
-    fprintf(stderr, "Usage: %s <filename.log>\n", prog);
+    fprintf(stderr, "Usage: %s [options] <filename.log>\n", prog);
+    fprintf(
+        stderr,
+        "\t-bsv: \t<Filename Base>: Exports the log to two files named "
+        "<Filename Base>-robot.bsv and\n\t\t\t<Filename Base>-frame.bsv and "
+        "Prevents launch of GUI\n");
+    fprintf(stderr, "\t-help: \tDisplays this help text\n");
     exit(1);
 }
 
 int main(int argc, char* argv[]) {
     QApplication app(argc, argv);
 
-    if (argc != 2) {
+    bool bsv = false;
+    char* bsvFilename;
+    char* logFilename;
+
+    // initial check for number of args
+    if (argc < 2 || argc > 4) {
         usage(argv[0]);
     }
 
+    // Process args
+    for (int i = 1; i < argc; ++i) {
+        const char* var = argv[i];
+
+        if (strcmp(var, "-bsv") == 0 && (i + 1 < argc)) {
+            bsv = true;
+            ++i;
+            bsvFilename = argv[i];
+        } else if (strcmp(var, "-help") == 0) {
+            usage(argv[0]);
+        } else {
+            logFilename = argv[i];
+        }
+    }
+
     LogViewer win;
 
-    win.readFrames(argv[1]);
-    win.showMaximized();
+    if (bsv) {
+        // Export BSV. Runs log_viewer in headless mode
+        if (win.exportBSV(logFilename, bsvFilename) == false) {
```

Comment:
What does the return value of `exportBSV()` mean?  Does `false` indicate success?